_posts/businessowl/2017-05-22-businessowl-turtle-hack

```diff
@@ -0,0 +1,13 @@
+---
+layout: post
+author: businessowl
+title: "Aaron Plocharczyk's Turtle Exercise"
+---
+
+Here's the program I'm embedding:
+
+    <iframe src="https://trinket.io/embed/python/18013b1bd3" width="100%" height="600" frameborder="0" marginwidth="0" marginheight="0" allowfullscreen></iframe>
+    
+I started out making a photoshop icon, and I was going to draw some background designs around it. Then I realized while drawing the box shape for the Photoshop icon that I could just draw a bunch of boxes to make a big grid across the screen. I could then fill in the boxes of that grid to display some retro pixel art, like a Mario character or something. Then I thought, why stop there? Actual pictures are nothing but a ton of pixels, so if I could extract color values from a real image, I could fill in the grid with those color values and have something that looks like a real image. I found a PHP function that extracts color values from images, and I just used nested for loops in PHP to write python code to draw a grid with those color values. My code is included for your viewing. It takes forever to draw, so I've included a link below with a preview image of the final result.
```

Comment:
Creative thought!

Revisiting this with what we're learning about loops, you'll be able to make a much more flexible program.  You could still use a PHP utility to make a text-based description of the image, but use a loop to go through the colors and build an image in turtle.

Perhaps this post would be inspiring: http://www.101computing.net/pixel-art-in-python/

In that post, users hand-code pixel art.  If you generated the text-based representation of the image required by the code examples above, you could then print any image in a pixelated style.

I like the persistence and creativity here.  Bringing in things you know from other programming language is good, of course, but one thing this hack lacks is much of your 'hand' in the code.  The Photoshop logo would've let you use far more of Turtle's capability and you'd have written far more actual Python.

That said, this gets high marks for creativity, precision, and persistence.